tools/report-converter/codechecker_report_converter/coccinelle/output_parser.py

```diff
@@ -0,0 +1,65 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import logging
+import os
+import re
+
+from ..output_parser import BaseParser, Message
+LOG = logging.getLogger('ReportConverter')
+
+
+class CoccinelleParser(BaseParser):
+    """
+    Parser for Coccinelle Output
+    """
+
+    def __init__(self, analyzer_result):
+        super(CoccinelleParser, self).__init__()
+
+        self.analyzer_result = analyzer_result
+
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            r'(?P<column>(\d+?)-(\d+?):)'
+            # r'(?P<bug>[\S ]+?:)'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*')
+
+    def parse_message(self, it, line):
+        """
+        Actual Parsing function for the given line
+        """
+        match = self.message_line_re.match(line)
+        if match is None:
+            return None, next(it)
+
+        matched_path = match.group('path')
+        matched_path = matched_path.split('./')[1]
```

Comment:
I actually do not think this path handling is actually to handle the path special, you could just have this work by running the importer in the _appropriate_ root directory of the repository, which is reasonable anyway when importing as you do need the sources at hand to import.

Reply:
Also, this functional change should be part of the implementation (if at all), not part of the commit that adds the test. Just because you state that "[test][fix]" and two bullet points does not make it a good commit to merge the two quite distinct points into one commit.

Reply:
Will put it as a part of the Parser commit :+1: